cluster-autoscaler/processors/nodes/pre_filtering_processor.go

```diff
@@ -48,6 +49,13 @@ func (n *PreFilteringScaleDownNodeProcessor) GetScaleDownCandidates(ctx *context
 	nodeGroupSize := utils.GetNodeGroupSizeMap(ctx.CloudProvider)
 
 	for _, node := range nodes {
+		if candidate, err := ctx.CloudProvider.IsNodeCandidateForDeletion(node); !candidate {
```

Comment:
I think we should first check the error and then the boolean

Otherwise folks have to be careful to always return true with the error

Reply:
I think this should be
```go
		if candidate, err := ctx.CloudProvider.IsNodeCandidateForScaleDown(node); err != nil {
			if err != cloudprovider.ErrNotImplemented {
				klog.Warningf("Error while checking if node is a candidate for deletion %s: %v", node.Name, err)
				continue
			}
		} else if !candidate {
			klog.V(5).Infof("Node %s is not a candidate for deletion (cloud provider determined)", node.Name)
			continue
		}
```
Otherwise we are hitting the `else if` branch if the func returns `ErrNotImplemented`